opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/ScopeManagerShim.java

```diff
@@ -18,22 +22,31 @@ public ScopeManagerShim(TelemetryInfo telemetryInfo) {
   @Override
   @SuppressWarnings("ReturnMissingNullable")
   public Span activeSpan() {
+    Context context = Context.current();
+    io.opentelemetry.api.trace.Span span = io.opentelemetry.api.trace.Span.fromContext(context);
+    SpanShim spanShim = context.get(SPAN_SHIM_KEY);
+
     // As OpenTracing simply returns null when no active instance is available,
     // we need to do map an invalid OpenTelemetry span to null here.
-    io.opentelemetry.api.trace.Span span = io.opentelemetry.api.trace.Span.current();
     if (!span.getSpanContext().isValid()) {
       return null;
     }
 
-    // TODO: Properly include the bagagge/distributedContext.
+    // If there's a SpanShim for the active Span, simply return it.
+    if (spanShim != null && spanShim.getSpan() == span) {
+      return spanShim;
+    }
+
+    // Span was activated from outside the Shim layer unfortunately.
     return new SpanShim(telemetryInfo(), span);
   }
 
   @Override
   @SuppressWarnings("MustBeClosedChecker")
   public Scope activate(Span span) {
     io.opentelemetry.api.trace.Span actualSpan = getActualSpan(span);
-    return new ScopeShim(actualSpan.makeCurrent());
+    Context context = Context.current().with(actualSpan).with(SPAN_SHIM_KEY, (SpanShim) span);
```

Comment:
If you implement `ImplicitContextKeyed`, I think you could have the implementation do the work of putting the span into the context as well.